### PR TITLE
Try harder to kill agent process in case of 'agent_expireafter' in cf-execd

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins with exotics, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4081/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4081/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4081/

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4188/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4188/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4188/

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4199/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4199/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4199/

File: cf-execd/cf-execd-runner.c [hunk 1]
```diff
@@ -263,11 +264,53 @@ void LocalExec(const ExecConfig *config)
...
+                /* kill(-pid) is actually kill(pgid=pid) and kill(pid, 0) just
+                 * checks if it's possible to send signals to pid (or pgid in
+                 * our case). Kill the whole process group if possible. */
+                if ((getpgid(pid_to_kill) == pid_to_kill)
+                    && (kill(-pid_to_kill, 0) == 0))
+                {
+                    pid_to_kill = -pid_to_kill;
+                }
+#endif
+                ProcessSignalTerminate(pid_to_kill);
```
Comment (olehermanse) [inline cf-execd/cf-execd-runner.c]: `pid_to_kill` was declared inside `#ifndef __MINGW32__`, and so it is undeclared here, on windows.

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4204/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4204/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4204/

**Comment by olehermanse** [comment]
Cherry-pick: https://github.com/cfengine/core/pull/3953

**Pull request merged**